File: src/main/java/at/yawk/wm/x/font/FontStyle.java
package at.yawk.wm.x.font;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.awt.*;
import lombok.Data;

/**
 * @author yawkat
 */
@Data
public class FontStyle {
    private Color color;
    private boolean bold = false;
    private boolean italic = false;
    private int size = 12;

    @JsonIgnore
    String getDescriptor() {
        StringBuilder builder = new StringBuilder(String.format("%06x", color.getRGB()));
        if (bold) { builder.append("-bold"); }
        if (italic) { builder.append("-italic"); }
        builder.append('-').append(size);
        return builder.toString();
    }

    public FontStyle withColor(Color color) {
        if (color.equals(this.color)) { return this; }
        FontStyle fs = new FontStyle();
        fs.color = color;
        fs.bold = bold;
        fs.italic = italic;
        fs.size = size;
        return fs;
    }
}

File: src/main/java/at/yawk/wm/style/FontStyle.kt
package at.yawk.wm.style

data class FontStyle(
    val family: String,
    val foreground: Color,
    val background: Color,
    val bold: Boolean,
    val italic: Boolean,
    val size: Int
) {
    fun getDescriptor(): String {
        val builder = StringBuilder(family)
        builder.append('-').append(java.lang.String.format("%06x", foreground.rgb))
        builder.append('-').append(java.lang.String.format("%06x", background.rgb))
        if (bold) {
            builder.append("-bold")
        }
        if (italic) {
            builder.append("-italic")
        }
        builder.append('-').append(size)
        return builder.toString()
    }
}